src/java.desktop/share/classes/java/awt/TextComponent.java

```diff
@@ -359,6 +359,11 @@ public void setBackground(Color c) {
         super.setBackground(c);
     }
 
+    void setBackground(Color c, boolean setByClient) {
+        backgroundSetByClientCode = setByClient;
```

Comment:
Assigning the passed value to `backgroundSetByClientCode` does not make sense to me: the code in `TextArea.setEditable` already depends on the value of `backgroundSetByClientCode`, and the field shouldn't change as the result.